test/semantic.test.js

```diff
@@ -42,3 +43,124 @@ describe('patch(version)', () => {
     expect(() => patch('not_version')).toThrowError(TypeError);
   });
 });
+
+describe('increment(version, major)', () => {
+  const testCases = [
+    // [ existing version, release type, identifier (`a`, `b`, `rc`, etc.), expected result ]
+
+    // Semantic Version numbers only.
+    ['0.0.0', 'major', undefined, '1.0.0'],
+    ['1.0.0', 'major', undefined, '2.0.0'],
+    ['1.0.1', 'major', undefined, '2.0.0'],
+    ['1.1.0', 'major', undefined, '2.0.0'],
+    ['0.0.0', 'minor', undefined, '0.1.0'],
+    ['1.0.0', 'minor', undefined, '1.1.0'],
+    ['1.0.1', 'minor', undefined, '1.1.0'],
+    ['1.1.0', 'minor', undefined, '1.2.0'],
+    ['0.0.0', 'patch', undefined, '0.0.1'],
+    ['1.0.0', 'patch', undefined, '1.0.1'],
+    ['1.0.1', 'patch', undefined, '1.0.2'],
+    ['1.1.0', 'patch', undefined, '1.1.1'],
+
+    // Extra release segments, as allowed by PEP440.
+    ['1.1.1.1', 'major', undefined, '2.0.0.0'],
+    ['1.1.1.1', 'minor', undefined, '1.2.0.0'],
+    ['1.1.1.1', 'patch', undefined, '1.1.2.0'],
```

Comment:
Should `inc` simply set the extra release segments to zero, or should it completely remove those release segments?

Reply:
`2.0.0.0` and `2` are the exact same version. the extra zeros could be preserved to maintain the same version format, projects usually keep the same release length when bumping the version.

Reply:
> the extra zeros could be preserved to maintain the same version format, projects usually keep the same release length when bumping the version.

Okay. So that was something I was wondering, and it's good to see confirmation.

Reply:
I have seen some projects jump around when releasing (e.g. from `1.5.1` to `2.0`) however I agree that for a bump function like this, keeping the same precision makes most sense.